.vib/ghost/cypress/cypress/integration/ghost_spec.js

```diff
@@ -0,0 +1,123 @@
+/// <reference types="cypress" />
+import { random, getPageUrlFromTitle, getUserFromEmail } from './utils';
+
+it('allows to log in and out', () => {
+  cy.login(Cypress.env('email'), Cypress.env('password'), false);
+  cy.contains('Retry').should('not.exist');
+  // Dashboard is hidden in the UI until correctly logged in
+  cy.contains('Dashboard').should('be.visible');
+
+  cy.get("div[class*='avatar']").click();
+  cy.contains('Sign out').click();
+  cy.contains('Sign in');
+});
+
+it('allows to create a new post', () => {
+  cy.login();
+  cy.visit('/ghost/#/posts');
+
+  cy.contains('New post').click();
+  cy.get('input[type="file"]').selectFile(
+    'cypress/fixtures/images/post_image.png',
+    { force: true }
+  );
+  cy.fixture('posts').then(($posts) => {
+    cy.get('textarea[placeholder="Post title"]').type(
+      `${$posts.newPost.title}-${random}`
+    );
+    cy.get('article').type(`${$posts.newPost.content}-${random}`);
+  });
+  // Publishing a post needs 3 steps
+  // Step 1: Open drop-down menu
+  cy.contains('Publish').click();
+  // Step 2: Select the option from the menu
+  cy.get('footer[class*=publishmenu]').within(() => {
+    cy.contains('Publish').click();
+  });
+  // Step 3: Confirmation pop-up
+  cy.get('div[class*=modal-content]').within(() => {
+    cy.contains('Publish').click();
+  });
+  // The notification exists but is hidden in the UI
+  cy.contains('Published').should('be.visible');
+
+  cy.visit('/');
+  cy.fixture('posts').then(($posts) => {
+    cy.contains(`${$posts.newPost.title}-${random}`);
+    cy.get(`img[alt='${$posts.newPost.title}-${random}']`);
+  });
+});
+
+it('allows to create a new page', () => {
+  cy.login();
+  cy.visit('/ghost/#/pages');
+
+  cy.contains('New page').click();
+  cy.fixture('pages').then(($pages) => {
+    cy.get('textarea[placeholder="Page title"]').type(
+      `${$pages.newPage.title}-${random}`
+    );
+    cy.get('article').type(`${$pages.newPage.content}-${random}`);
+  });
+  // Publishing a page needs 2 steps
+  // Step 1: Open drop down-menu
+  cy.contains('Publish').click();
+  // Step 2: Select the option from the menu
+  cy.get('footer[class*=publishmenu]').within(() => {
+    cy.contains('Publish').click();
+  });
+  // The notification exists but is hidden in the UI
+  cy.contains('Published').should('be.visible');
+
+  cy.fixture('pages').then(($pages) => {
+    cy.visit(getPageUrlFromTitle(`${$pages.newPage.title}-${random}`));
+    cy.contains(`${$pages.newPage.title}-${random}`);
+    cy.contains(`${$pages.newPage.content}-${random}`);
+  });
+});
+
+it('allows to import members', () => {
+  cy.login();
+  cy.visit('/ghost/#/members/import');
+
```

Comment:
Okay, agreed. If there is an explicit agreement on trying to avoid `be.visible/exists` when possible we should trust the author, and hence not need these kinds of comments.